### PR TITLE
Admin dashboard with initial content

### PR DESCRIPTION
**Pull request by loic425**
Capture d’écran 2019-07-25 à 11 18 27

### CODE REVIEW

File: src/Repository/CustomerRepository.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+class CustomerRepository extends EntityRepository
+{
+    public function countCustomers(): int
+    {
+        return (int) $this->createQueryBuilder('o')
+            ->select('COUNT(o.id)')
+            ->getQuery()
+            ->getSingleScalarResult()
+            ;
```
Comment (lchrusciel) [inline src/Repository/CustomerRepository.php]: Semicolon here and in the method below is written differently :) We could improve consistency here. Can you run easy coding standard?

File: src/Controller/DashboardController.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\Dashboard\DashboardStatisticsProvider;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+
+class DashboardController extends AbstractController
```
Comment (lchrusciel) [inline src/Controller/DashboardController.php]: As we are using proper DI in this controller I would vote for removing AbstractController inheritance and just inject twig renderer. 

Also, we could finalize this service. Ref. http://ocramius.github.io/blog/when-to-declare-classes-final/

Comment (lchrusciel) [reply src/Controller/DashboardController.php]: Same for all of services

File: spec/App/Controller/DashboardControllerSpec.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+namespace spec\App\Controller;
+
+use App\Controller\DashboardController;
+use App\Dashboard\DashboardStatistics;
+use App\Dashboard\DashboardStatisticsProvider;
+use PhpSpec\ObjectBehavior;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Templating\EngineInterface;
+
+class DashboardControllerSpec extends ObjectBehavior
```
Comment (lchrusciel) [inline spec/App/Controller/DashboardControllerSpec.php]: There is no point in writing specs for controller. Let's test it with behat only

File: spec/App/Controller/DashboardControllerSpec.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+{
+    function let(DashboardStatisticsProvider $statisticsProvider, ContainerInterface $container): void
+    {
+        $this->beConstructedWith($statisticsProvider);
+        $this->setContainer($container);
```
Comment (lchrusciel) [inline spec/App/Controller/DashboardControllerSpec.php]: That's why I think, that depending on AbstractController is a bad thing

File: config/services.yaml [hunk 5]
```diff
@@ -8,6 +8,17 @@ services:
...
         autoconfigure: true
 
+    # controllers are imported separately to make sure services can be injected
+    # as action arguments even if you don't extend any base controller class
+    App\Controller\:
+        resource: '../src/Controller'
+        autowire: true
+        tags: ['controller.service_arguments']
+
+    App\Dashboard\:
```
Comment (lchrusciel) [inline config/services.yaml]: Isn't it resolved out-of-the-box?

Comment (lchrusciel) [reply config/services.yaml]: And just quick thought. I would say, that people are used to having imports at the beginning of the file

File: config/routes/backend/partial/customer.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+sylius_backend_partial_customer_latest:
```
Comment (lchrusciel) [inline config/routes/backend/partial/customer.yaml]: Why do you prefix it with sylius?

```suggestion
monofony_backend_partial_customer_latest:
```

We have a new brand here :)

File: config/routes/backend/partial.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+sylius_backend_partial_customer:
+    resource: partial/customer.yaml
+    prefix: /customers
+
```
Comment (lchrusciel) [inline config/routes/backend/partial.yaml]: Redundant

File: config/routes/backend/partial/customer.yaml (hunk 6 above)
Comment (loic425) [reply config/routes/backend/partial/customer.yaml]: I follow SyliusResourceBundle routes generator to be consistent.
if i's a Sylius resource, I prefix by "sylius_" and if it's a custom resource I prefix by "app_".

File: config/services.yaml (hunk 5 above)
Comment (loic425) [reply config/services.yaml]: > Isn't it resolved out-of-the-box?

Not yet, I need to work on that to be closer to Symfony skeleton.

File: src/Repository/CustomerRepository.php (hunk 1 above)
Comment (loic425) [reply src/Repository/CustomerRepository.php]: I try to run coding standard, but no rules are defined about it.

File: config/services.yaml (hunk 5 above)
Comment (loic425) [reply config/services.yaml]: > And just quick thought. I would say, that people are used to having imports at the beginning of the file

Yes, I moved these imports on top.

**Pull request merged**